### PR TITLE
GUM21: New/scaffold profile

### PR DESCRIPTION
**Pull request by Ihore4ik**

### CODE REVIEW

File: src/screens/Profile/index.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import {View,Text, StyleSheet} from 'react-native';
```
Comment (yevhenprybytok) [inline src/screens/Profile/index.tsx]:
```suggestion
import { View, Text, StyleSheet } from 'react-native';
```

File: src/screens/Profile/index.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import {Button, Divider} from 'react-native-paper';
+import {SafeAreaView} from 'react-native-safe-area-context';
+
+type IHomeScreenProps = Record<string, any>;
+type State = Array<{ id: number, text: string, route: JSX.Element }>
+const ProfileScreen: React.FC<IHomeScreenProps> = () => {
```
Comment (yevhenprybytok) [inline src/screens/Profile/index.tsx]:
```suggestion

const ProfileScreen: React.FC<IHomeScreenProps> = () => {
```

File: src/screens/Profile/index.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+			text: 'Qualities',
+			route: <Text>Show Qualities component</Text>
+		},
+		{
+			id: 2,
+			text: 'Interests',
+			route: <Text>Show Interests component</Text>
+		},
+	]);
+	const [active, setActive] = React.useState(0);
```
Comment (yevhenprybytok) [inline src/screens/Profile/index.tsx]:
```suggestion
	
	const [active, setActive] = React.useState(0);
```

File: src/screens/Profile/index.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	const handleClick = (ind: number) => setActive(ind);
```
Comment (yevhenprybytok) [inline src/screens/Profile/index.tsx]:
```suggestion
	
	const handleClick = (ind: number) => setActive(ind);
```

File: src/screens/Profile/index.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	return (
+		<SafeAreaView>
+			<View style={styles.container}>
+				<View style={styles.navbar}>
+					<Text style={styles.title}>User profile</Text>
+					<View style={styles.buttons}>
+						{items.map((btn, i) => {
```
Comment (yevhenprybytok) [inline src/screens/Profile/index.tsx]:
```suggestion
						{items.map((item, index) => {
```

File: src/screens/Profile/index.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+							return (
+								<Button
+									key={i}
+									style={active === i ? styles.active : styles.btn}
+									mode={active === i ? 'contained' : 'text'}
+									color={active === i ? '' : '#000'}
```
Comment (yevhenprybytok) [inline src/screens/Profile/index.tsx]: Move colors into consts

File: src/screens/Profile/index.tsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+					}
+				</View>
+			</View>
+		</SafeAreaView>
+	);
+};
+
+export default ProfileScreen;
+
+const styles = StyleSheet.create({
```
Comment (yevhenprybytok) [inline src/screens/Profile/index.tsx]: Move styles away to another file

**Review by yevhenprybytok — CHANGES_REQUESTED** [review]

File: .prettierrc.js [hunk 8]
```diff
@@ -9,4 +9,4 @@ module.exports = {
   tabWidth: 2,
   useTabs: false,
   endOfLine: 'lf',
-};
+}
```
Comment (R-Mielamud) [inline .prettierrc.js]: We have `semi: true` right here, so return semicolon back.

File: android/app/src/main/java/com/growupmobile/MainApplication.java [hunk 9]
```diff
@@ -4,6 +4,7 @@
 import android.content.Context;
 import com.facebook.react.PackageList;
 import com.facebook.react.ReactApplication;
+import com.reactnativepagerview.PagerViewPackage;
```
Comment (R-Mielamud) [inline android/app/src/main/java/com/growupmobile/MainApplication.java]: Remove this import and run `npx react-native link` in terminal/cmdline

File: src/screens/profile/components/navbar/navbar.tsx [hunk 10]
```diff
@@ -0,0 +1,73 @@
+import React, { useRef } from 'react';
+import { Dimensions, ScrollView, View } from 'react-native';
+
+import { MainButton, Text } from 'src/components';
+import { AppColor, ButtonMode } from 'src/common/enums';
+
+import styles from './styles';
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/navbar.tsx]: Remove new line

File: src/screens/profile/components/navbar/navbar.tsx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+type State = Array<{ id: number; text: string }>;
+
+interface Props {
+  active: number;
+  handleClick: (a: number) => void;
+}
+
+const WIDTH = Dimensions.get('screen').width * 0.8;
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/navbar.tsx]: Add more meaningful name and extract to `src/constants`

File: src/screens/profile/components/navbar/navbar.tsx [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+const BUTTON_WIDTH = Math.ceil(WIDTH / 3);
+
+const Navbar = ({ active, handleClick }: Props) => {
+  const scrollRef = useRef<ScrollView>(null);
+
+  scrollRef.current?.scrollTo({
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/navbar.tsx]: Wrap in:
```typescript
useEffect(() => {
  ...
}, [active]);
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+    x: active * BUTTON_WIDTH,
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/navbar.tsx]: Extract this to variable with meaningful name or even comment.

File: src/screens/profile/components/navbar/navbar.tsx [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+    y: 0,
+    animated: true,
+  });
+
+  const [items] = React.useState<State>([
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/navbar.tsx]: Just `useState<State>` (and import this function like this: `import React, { useState, useEffect } from 'react'`)

File: src/screens/profile/components/navbar/navbar.tsx [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+
+  return (
+    <View style={styles.navbar}>
+      <Text style={styles.title}>User profile</Text>
+      <View style={styles.buttons}>
+        <ScrollView horizontal={true} ref={scrollRef}>
+          {items.map((item, index) => {
+            return (
+              <MainButton
+                key={item.text}
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/navbar.tsx]: Not a good idea to use `text` in key. Better use `id`

File: src/screens/profile/components/navbar/navbar.tsx [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+    {
+      id: 2,
+      text: 'Interests',
+    },
+    {
+      id: 3,
+      text: 'Skills',
+    },
+    {
+      id: 3,
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/navbar.tsx]: Duplicate `id`'s

File: src/screens/profile/components/navbar/navbar.tsx [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+        <ScrollView horizontal={true} ref={scrollRef}>
+          {items.map((item, index) => {
+            return (
+              <MainButton
+                key={item.text}
+                style={active === index ? styles.active : styles.btn}
+                mode={active === index ? ButtonMode.CONTAINED : ButtonMode.TEXT}
+                color={active === index ? '' : AppColor.BLACK}
+                onPress={() => handleClick(index)}
+                compact={true}
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/navbar.tsx]: Swap these 2 lines

File: src/screens/profile/components/navbar/navbar.tsx [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+    },
+  ]);
+
+  return (
+    <View style={styles.navbar}>
+      <Text style={styles.title}>User profile</Text>
+      <View style={styles.buttons}>
+        <ScrollView horizontal={true} ref={scrollRef}>
+          {items.map((item, index) => {
+            return (
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/navbar.tsx]: Unuseful return statement. Just
```typescript
{items.map((item, index) => (
  <MainButton ... />
)}
```

File: src/screens/profile/components/navbar/styles.ts [hunk 19]
```diff
@@ -0,0 +1,43 @@
+import { Dimensions, StyleSheet } from 'react-native';
+
+import { AppColor } from 'src/common/enums';
+
+const WIDTH = Dimensions.get('screen').width * 0.8;
+const BUTTON_WIDTH = Math.ceil(WIDTH / 3);
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/styles.ts]: You also use them in component, so extract to `src/constants` and add more meaningful name

File: src/screens/profile/profile.tsx [hunk 20]
```diff
@@ -1,17 +1,26 @@
...
 import addActions from './add-actions';
-import styles from './styles';
+
+import { styles } from './styles';
 
 type IProfileScreenProps = Record<string, never>;
 
 const ProfileScreen: React.FC<IProfileScreenProps> = () => {
   const [addMenuOpen, setAddMenuOpen] = useState<boolean>(false);
+  const [active, setActive] = React.useState<number>(0);
```
Comment (R-Mielamud) [inline src/screens/profile/profile.tsx]: `useState` instead of `React.useState`

File: src/screens/profile/profile.tsx [hunk 21]
```diff
@@ -1,17 +1,26 @@
...
+  const refPage = useRef<PagerView>(null);
 
+  const handleClick = useCallback((ind: number) => {
```
Comment (R-Mielamud) [inline src/screens/profile/profile.tsx]: `index: number`

File: src/screens/profile/profile.tsx [hunk 22]
```diff
@@ -1,17 +1,26 @@
...
+import PagerView from 'react-native-pager-view';
 
+import { Navbar } from 'src/screens/profile/components/navbar';
 import { Text } from 'src/components';
 import addActions from './add-actions';
-import styles from './styles';
+
+import { styles } from './styles';
 
 type IProfileScreenProps = Record<string, never>;
```
Comment (R-Mielamud) [inline src/screens/profile/profile.tsx]: This is not an interface. Just `ProfileScreenProps`

File: src/screens/profile/profile.tsx [hunk 23]
```diff
@@ -39,8 +48,36 @@ const ProfileScreen: React.FC<IProfileScreenProps> = () => {
...
+        <Navbar active={active} handleClick={handleClick} />
+        <Divider />
+        <View style={styles.content}>
+          <PagerView
+            initialPage={0}
+            ref={refPage}
+            onPageSelected={({ nativeEvent }) =>
+              setActive(nativeEvent.position)
+            }
+            orientation={'horizontal'}
```
Comment (R-Mielamud) [inline src/screens/profile/profile.tsx]: `orientation="horizontal"`

File: src/screens/profile/styles.ts [hunk 24]
```diff
@@ -1,9 +1,32 @@
 import { StyleSheet } from 'react-native';
 
-const styles = StyleSheet.create({
+import { AppColor } from 'src/common/enums';
+
+export const styles = StyleSheet.create({
```
Comment (R-Mielamud) [inline src/screens/profile/styles.ts]: export default

File: src/screens/profile/styles.ts [hunk 25]
```diff
@@ -1,9 +1,32 @@
...
+  },
+  swiperItem: {
+    width: '100%',
+    height: '100%',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
 });
-
-export default styles;
```
Comment (R-Mielamud) [inline src/screens/profile/styles.ts]: Don't remove this

**Review by R-Mielamud — CHANGES_REQUESTED** [review]

File: src/screens/profile/components/navbar/styles.ts [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+    margin: 10,
+    width: PROFILE_NAVBAR_WIDTH,
+    alignSelf: 'center',
+    borderRadius: 7,
+  },
+  title: {
+    color: AppColor.BLACK,
+    fontSize: 17,
+    textAlign: 'center',
+    fontFamily: 'NunitoSans-Regular',
```
Comment (R-Mielamud) [inline src/screens/profile/components/navbar/styles.ts]: unnecessary line

**Review by R-Mielamud — COMMENTED** [review]
Minor fixes

File: src/screens/profile/components/navbar/styles.ts (hunk 26 above)
Comment (Ihore4ik) [reply src/screens/profile/components/navbar/styles.ts]: Fix in [8474675](https://github.com/BinaryStudioAcademy/bsa-winter-2022-growup-mobile/pull/9/commits/8474675de7e43321d635cccf187b3c55644409af)

**Review by R-Mielamud — APPROVED** [review]

File: src/screens/profile/components/navbar/navbar.tsx [hunk 27]
```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { ScrollView, View } from 'react-native';
+
+import { MainButton, Text } from 'src/components';
+import { AppColor, ButtonMode } from 'src/common/enums';
+import { BUTTON_PROFILE_NAVBAR_WIDTH } from 'src/common/constants';
+import styles from './styles';
+
+type State = Array<{ id: number; text: string }>;
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]:
```suggestion
type Item = { 
  id: number; 
  text: string; 
};
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 28]
```diff
@@ -0,0 +1,71 @@
...
+
+interface Props {
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]:
```suggestion
type NavbarProps {
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 29]
```diff
@@ -0,0 +1,71 @@
...
+  active: number;
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]:
```suggestion
  activeIndex: number;
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 30]
```diff
@@ -0,0 +1,71 @@
...
+  handleClick: (a: number) => void;
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]: Callbacks passed to props must be named via on.
```suggestion
  onClick: (index: number) => void;
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 31]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+const Navbar = ({ active, handleClick }: Props) => {
+  const scrollRef = useRef<ScrollView>(null);
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]:
```suggestion
  const scrollViewRef = useRef<ScrollView>(null);
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 32]
```diff
@@ -0,0 +1,71 @@
...
+
+  useEffect(() => {
+    const scrollToPosition = active * BUTTON_PROFILE_NAVBAR_WIDTH;
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]: We should use nouns for variables and verbs for functions
```suggestion
    const xOffset = active * BUTTON_PROFILE_NAVBAR_WIDTH;
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 33]
```diff
@@ -0,0 +1,71 @@
...
+    scrollRef.current?.scrollTo({
+      x: scrollToPosition,
+      y: 0,
+      animated: true,
+    });
+  }, [active]);
+
+  const [items] = useState<State>([
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]:
```suggestion
  const [items] = useState<Item[]>([
```

File: src/screens/profile/components/navbar/styles.ts [hunk 34]
```diff
@@ -0,0 +1,42 @@
...
+    margin: 10,
+    width: PROFILE_NAVBAR_WIDTH,
+    alignSelf: 'center',
+    borderRadius: 7,
+  },
+  title: {
+    fontSize: 17,
+    textAlign: 'center',
+  },
+  btn: {
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/styles.ts]:
```suggestion
  button: {
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 35]
```diff
@@ -0,0 +1,71 @@
...
+
+  return (
+    <View style={styles.navbar}>
+      <Text style={styles.title}>User profile</Text>
+      <View style={styles.buttons}>
+        <ScrollView horizontal={true} ref={scrollRef}>
+          {items.map((item, index) => (
+            <MainButton
+              key={item.id}
+              style={active === index ? styles.active : styles.btn}
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]:
```suggestion
              style={[styles.button, active && styles.buttonActive]}
```

File: src/screens/profile/components/navbar/styles.ts [hunk 36]
```diff
@@ -0,0 +1,42 @@
...
+    flex: 1,
+    width: BUTTON_PROFILE_NAVBAR_WIDTH,
+    borderRadius: 0,
+  },
+  active: {
+    width: BUTTON_PROFILE_NAVBAR_WIDTH,
+    borderRadius: 7,
+    backgroundColor: AppColor.ACCENT,
+    flex: 1,
+  },
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/styles.ts]:
```suggestion
  buttonActive: {
    borderRadius: 7,
    backgroundColor: AppColor.ACCENT,
  },
```

File: src/screens/profile/profile.tsx [hunk 37]
```diff
@@ -1,19 +1,28 @@
...
+import PagerView from 'react-native-pager-view';
 
+import { Navbar } from 'src/screens/profile/components/navbar';
 import { ProfileRoute } from 'src/common/enums';
 import { Text } from 'src/components';
 import addActions from './add-actions';
 import styles from './styles';
 
-type IProfileScreenProps = Record<string, never>;
+type ProfileScreenProps = Record<string, never>;
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]: Don't remove I, we use this style for all screens
```suggestion
type IProfileScreenProps = Record<string, never>;
```

File: src/screens/profile/profile.tsx [hunk 38]
```diff
@@ -1,19 +1,28 @@
...
 
-const ProfileScreen: React.FC<IProfileScreenProps> = () => {
+const ProfileScreen: React.FC<ProfileScreenProps> = () => {
   const navigation = useNavigation();
   const [addMenuOpen, setAddMenuOpen] = useState<boolean>(false);
+  const [active, setActive] = React.useState<number>(0);
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]:
```suggestion
  const [activeIndex, setActiveIndex] = React.useState(0);
```

File: src/screens/profile/profile.tsx [hunk 39]
```diff
@@ -1,19 +1,28 @@
...
 import addActions from './add-actions';
 import styles from './styles';
 
-type IProfileScreenProps = Record<string, never>;
+type ProfileScreenProps = Record<string, never>;
 
-const ProfileScreen: React.FC<IProfileScreenProps> = () => {
+const ProfileScreen: React.FC<ProfileScreenProps> = () => {
   const navigation = useNavigation();
   const [addMenuOpen, setAddMenuOpen] = useState<boolean>(false);
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]: Let's avoid unnecessary types
```suggestion
  const [addMenuOpen, setAddMenuOpen] = useState(false);
```

File: src/screens/profile/profile.tsx [hunk 40]
```diff
@@ -1,19 +1,28 @@
...
+  const [active, setActive] = React.useState<number>(0);
+  const refPage = useRef<PagerView>(null);
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]:
```suggestion
  const pagerViewRef = useRef<PagerView>(null);
```

File: src/screens/profile/profile.tsx [hunk 41]
```diff
@@ -1,19 +1,28 @@
...
+
+  const handleClick = useCallback((index: number) => {
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]: It's better not to use useCallback here. Please take a look at this [article](https://kentcdodds.com/blog/usememo-and-usecallback) when you have time

File: src/screens/profile/profile.tsx [hunk 42]
```diff
@@ -49,8 +58,36 @@ const ProfileScreen: React.FC<IProfileScreenProps> = () => {
...
+      <View style={styles.container}>
+        <Navbar active={active} handleClick={handleClick} />
+        <Divider />
+        <View style={styles.content}>
+          <PagerView
+            initialPage={0}
+            ref={refPage}
+            onPageSelected={({ nativeEvent }) =>
+              setActive(nativeEvent.position)
+            }
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]: Let's move it callback like handlePageSelect

File: src/screens/profile/styles.ts [hunk 43]
```diff
@@ -1,9 +1,34 @@
...
+    alignItems: 'center',
+  },
+  content: {
+    flex: 1,
+    width: '100%',
+    height: '100%',
+    paddingTop: 15,
+    backgroundColor: AppColor.WHITE,
+  },
+  swipeWrapper: {
```
Comment (liza-veis) [inline src/screens/profile/styles.ts]:
```suggestion
  swiperWrapper: {
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: src/screens/profile/components/navbar/navbar.tsx (hunk 35 above)
Comment (Ihore4ik) [reply src/screens/profile/components/navbar/navbar.tsx]: It works but like this: `style={[styles.button,  activeIndex === index && styles.buttonActive ] }`

File: src/screens/profile/styles.ts [hunk 44]
```diff
@@ -1,9 +1,31 @@
...
 
+import { AppColor } from 'src/common/enums';
+
 const styles = StyleSheet.create({
   fullHeight: {
     height: '100%',
   },
+  container: {
+    flex: 1,
+    alignItems: 'center',
```
Comment (liza-veis) [inline src/screens/profile/styles.ts]: We don't need alignItems here

File: src/screens/profile/styles.ts [hunk 45]
```diff
@@ -1,9 +1,31 @@
...
+  },
+  content: {
+    width: '100%',
+    height: '100%',
```
Comment (liza-veis) [inline src/screens/profile/styles.ts]: if you remove align-items above, you can do this:
```suggestion
    flex: 1,
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 46]
```diff
@@ -0,0 +1,79 @@
...
+    },
+    {
+      id: 3,
+      text: 'Skills',
+    },
+    {
+      id: 4,
+      text: 'Education',
+    },
+  ]);
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]: I think we can reduce it to just array with strings. In profile tsx create constant like NAVBAR_ITEMS=['Qualities', 'Interests', ...] and pass here via props

File: src/common/constants/profile-screen.ts [hunk 47]
```diff
@@ -0,0 +1,6 @@
+import { Dimensions } from 'react-native';
+
+const PROFILE_NAVBAR_WIDTH = Dimensions.get('screen').width * 0.8;
+const BUTTON_PROFILE_NAVBAR_WIDTH = Math.ceil(PROFILE_NAVBAR_WIDTH / 3);
```
Comment (liza-veis) [inline src/common/constants/profile-screen.ts]: Rename this file to app.ts and leave here only screen width
```suggestion
const SCREEN_WIDTH = Dimensions.get('screen').width;
```

Let's put button width constant in navbar/constants.ts file:
```
import { SCREEN_WIDTH } from 'src/common/constants';
import { DEFAULT_SCREEN_PADDING } from 'src/styles';

const NAVBAR_BUTTON_WIDTH = Math.ceil(
  (SCREEN_WIDTH - DEFAULT_SCREEN_PADDING * 2) / 3
);

export { NAVBAR_BUTTON_WIDTH };
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 48]
```diff
@@ -0,0 +1,79 @@
...
+
+  return (
+    <View style={styles.navbar}>
+      <Text style={styles.title}>User profile</Text>
+      <View style={styles.buttons}>
+        <ScrollView horizontal={true} ref={scrollViewRef}>
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]: Let's remove this View and pass styles to scrollView
```suggestion
      <ScrollView
        horizontal={true}
        ref={scrollViewRef}
      >
```

File: src/screens/profile/components/navbar/styles.ts [hunk 49]
```diff
@@ -0,0 +1,40 @@
...
+  BUTTON_PROFILE_NAVBAR_WIDTH,
+  PROFILE_NAVBAR_WIDTH,
+} from 'src/common/constants';
+
+const styles = StyleSheet.create({
+  navbar: {
+    paddingTop: 10,
+    justifyContent: 'space-between',
+    width: '100%',
+    height: 93,
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/styles.ts]:
```suggestion
    paddingHorizontal: DEFAULT_SCREEN_PADDING,
    paddingVertical: 10,
```

File: src/screens/profile/components/navbar/styles.ts [hunk 50]
```diff
@@ -0,0 +1,40 @@
...
+    backgroundColor: AppColor.INPUT_BACKGROUND,
+  },
+  buttons: {
+    flexDirection: 'row',
+    backgroundColor: AppColor.WHITE,
+    margin: 10,
+    width: PROFILE_NAVBAR_WIDTH,
+    alignSelf: 'center',
+    borderRadius: 7,
+  },
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/styles.ts]: We don't need this style
```suggestion
```

File: src/screens/profile/components/navbar/styles.ts [hunk 51]
```diff
@@ -0,0 +1,40 @@
...
+  title: {
+    fontSize: 17,
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/styles.ts]:
```suggestion
    fontSize: 17,
    marginBottom: 10,
```

File: src/screens/profile/components/navbar/styles.ts [hunk 52]
```diff
@@ -0,0 +1,40 @@
...
+    textAlign: 'center',
+  },
+  button: {
+    flex: 1,
+    width: BUTTON_PROFILE_NAVBAR_WIDTH,
+    borderRadius: 0,
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/styles.ts]: Let's set borderRadius 7 here and remove flex
```suggestion
    width: BUTTON_PROFILE_NAVBAR_WIDTH,
    borderRadius: 7,
```

File: src/screens/profile/components/navbar/styles.ts [hunk 53]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  activeButton: {
+    borderRadius: 7,
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/styles.ts]:
```suggestion
```

File: src/screens/profile/profile.tsx [hunk 54]
```diff
@@ -13,7 +17,14 @@ type IProfileScreenProps = Record<string, never>;
 
 const ProfileScreen: React.FC<IProfileScreenProps> = () => {
   const navigation = useNavigation();
-  const [addMenuOpen, setAddMenuOpen] = useState<boolean>(false);
+  const [addMenuOpen, setAddMenuOpen] = useState(false);
+  const [activeIndex, setActiveIndex] = useState(0);
+  const refViewPage = useRef<PagerView>(null);
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]:
```suggestion
  const pagerRef = useRef<PagerView>(null);
```

File: package.json [hunk 55]
```diff
@@ -22,6 +22,7 @@
     "react-native": "0.67.2",
     "react-native-config": "^1.4.5",
     "react-native-encrypted-storage": "^4.0.2",
+    "react-native-pager-view": "^5.4.9",
```
Comment (liza-veis) [inline package.json]: We already have react-native-pager-view

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: src/screens/profile/components/navbar/navbar.tsx [hunk 56]
```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useRef } from 'react';
+import { ScrollView, View } from 'react-native';
+
+import { MainButton, Text } from 'src/components';
+import { AppColor, ButtonMode } from 'src/common/enums';
+import { NAVBAR_BUTTON_WIDTH } from 'src/screens/profile/components/navbar/constants';
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]:
```suggestion
import { NAVBAR_BUTTON_WIDTH } from './constants';
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 57]
```diff
@@ -0,0 +1,53 @@
...
+import styles from './styles';
+
+type NavbarProps = {
+  activeIndex: number;
+  onClick: (index: number) => void;
+  items: string[];
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]: Let's keep methods after properties
```suggestion
  items: string[];
  onClick: (index: number) => void;
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 58]
```diff
@@ -0,0 +1,53 @@
...
+    });
+  }, [activeIndex]);
+
+  return (
+    <View style={styles.navbar}>
+      <Text style={styles.title}>User profile</Text>
+      <ScrollView horizontal={true} ref={scrollViewRef}>
+        {items.map((item, index) => (
+          <MainButton
+            key={index}
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/navbar.tsx]: It's better to use item for key here
```suggestion
            key={item}
```

File: src/screens/profile/components/navbar/styles.ts [hunk 59]
```diff
@@ -0,0 +1,28 @@
+import { StyleSheet } from 'react-native';
+
+import { AppColor } from 'src/common/enums';
+import { DEFAULT_SCREEN_PADDING } from 'src/styles';
+import { NAVBAR_BUTTON_WIDTH } from 'src/screens/profile/components/navbar/constants';
```
Comment (liza-veis) [inline src/screens/profile/components/navbar/styles.ts]:
```suggestion
import { NAVBAR_BUTTON_WIDTH } from './constants';
```

File: src/screens/profile/profile.tsx [hunk 60]
```diff
@@ -13,7 +17,21 @@ type IProfileScreenProps = Record<string, never>;
 
 const ProfileScreen: React.FC<IProfileScreenProps> = () => {
   const navigation = useNavigation();
-  const [addMenuOpen, setAddMenuOpen] = useState<boolean>(false);
+  const [addMenuOpen, setAddMenuOpen] = useState(false);
+  const [activeIndex, setActiveIndex] = useState(0);
+  const pagerRef = useRef<PagerView>(null);
+  const NAVBAR_ITEMS = [
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]: Move NAVBAR_ITEMS outside the component

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: src/screens/profile/components/navbar/styles.ts (hunk 59 above)
Comment (Ihore4ik) [reply src/screens/profile/components/navbar/styles.ts]: Auto import, fix in [357f930](https://github.com/BinaryStudioAcademy/bsa-winter-2022-growup-mobile/pull/9/commits/357f93093cf6dd705105610e61633e33f8b1e92b)

File: src/screens/profile/profile.tsx (hunk 60 above)
Comment (Ihore4ik) [reply src/screens/profile/profile.tsx]: Fix in [357f930](https://github.com/BinaryStudioAcademy/bsa-winter-2022-growup-mobile/pull/9/commits/357f93093cf6dd705105610e61633e33f8b1e92b)

**Review by liza-veis — APPROVED** [review]

**Review by katyakostikova — APPROVED** [review]

File: ios/Podfile [hunk 61]
```diff
@@ -12,6 +12,8 @@ target 'growupmobile' do
     :hermes_enabled => false
   )
 
+  pod 'react-native-pager-view', :path => '../node_modules/react-native-pager-view'
```
Comment (AleksandrTermenzhy) [inline ios/Podfile]: Do we really need this manual linking?
I assume we already have this library installed and being used

File: src/screens/profile/components/navbar/constants.ts [hunk 62]
```diff
@@ -0,0 +1,8 @@
+import { SCREEN_WIDTH } from 'src/common/constants';
+import { DEFAULT_SCREEN_PADDING } from 'src/styles';
+
+const NAVBAR_BUTTON_WIDTH = Math.ceil(
+  (SCREEN_WIDTH - DEFAULT_SCREEN_PADDING * 2) / 3
```
Comment (AleksandrTermenzhy) [inline src/screens/profile/components/navbar/constants.ts:5]: Why do we need this? And what we'll do if there are more than 3 buttons or with long-name-button?

File: src/screens/profile/components/navbar/navbar.tsx [hunk 63]
```diff
@@ -0,0 +1,53 @@
...
+    scrollViewRef.current?.scrollTo({
+      x: xOffset,
+      y: 0,
+      animated: true,
+    });
+  }, [activeIndex]);
+
+  return (
+    <View style={styles.navbar}>
+      <Text style={styles.title}>User profile</Text>
```
Comment (AleksandrTermenzhy) [inline src/screens/profile/components/navbar/navbar.tsx]: This is screen title. We don't need it here.

File: src/screens/profile/components/navbar/navbar.tsx [hunk 64]
```diff
@@ -0,0 +1,57 @@
...
+          <MainButton
+            key={item}
+            style={[
+              styles.button,
+              activeIndex === index && styles.activeButton,
+            ]}
+            mode={
+              activeIndex === index ? ButtonMode.CONTAINED : ButtonMode.TEXT
+            }
+            color={activeIndex === index ? '' : AppColor.BLACK}
```
Comment (AleksandrTermenzhy) [inline src/screens/profile/components/navbar/navbar.tsx]:
```suggestion
            color={activeIndex === index ? AppColor.WHITE : AppColor.BLACK}
```

File: src/screens/profile/components/navbar/navbar.tsx [hunk 65]
```diff
@@ -0,0 +1,57 @@
...
+  const isButtonsScrolling = items.length < 4;
+
+  useEffect(() => {
+    const xOffset = activeIndex * NAVBAR_BUTTON_WIDTH;
+    scrollViewRef.current?.scrollTo({
+      x: xOffset,
+      y: 0,
+      animated: true,
+    });
+  }, [activeIndex]);
```
Comment (AleksandrTermenzhy) [inline src/screens/profile/components/navbar/navbar.tsx]:
```suggestion
  }, [activeIndex, scrollViewRef]);
```

**Review by AleksandrTermenzhy — APPROVED** [review]

**Review by yevhenprybytok — APPROVED** [review]

**Pull request merged**